.github/workflows/generator_generic_slsa3.yml

```diff
@@ -141,3 +154,40 @@ jobs:
           path: "${{ steps.sign-prov.outputs.attestation-name }}"
           if-no-files-found: error
           retention-days: 5
+
+  # create-release creates a release and uploads provenance to the release if
+  # `upload-to-release` is true.
+  create-release:
+    outputs:
+      release-id: ${{ steps.release.outputs.id }}
+    runs-on: ubuntu-latest
+    needs: [generator]
+    permissions:
+      contents: write
+    if: startsWith(github.ref, 'refs/tags/') && inputs.upload-to-release == true
+    steps:
+      # Verify provenance hash.
+      - uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+        with:
+          name: ${{ needs.generator.outputs.attestation-name }}
+      - name: Verify provenance hash
+        env:
+          UNTRUSTED_PROVENANCE_NAME: "${{ needs.generator.outputs.attestation-name }}"
+          UNTRUSTED_PROVENANCE_HASH: "${{ needs.generator.outputs.attestation-sha256 }}"
+        run: |
+          set -euo pipefail
+
+          echo "hash of provenance $UNTRUSTED_PROVENANCE_NAME should be $UNTRUSTED_PROVENANCE_HASH"
+
+          COMPUTED_HASH=$(sha256sum "$UNTRUSTED_PROVENANCE_NAME" | awk '{print $1}')
+          echo "provenance hash computed is $COMPUTED_HASH"
+
+          # Compare hashes. Explicit exit to be safe.
+          echo "$UNTRUSTED_PROVENANCE_HASH $UNTRUSTED_PROVENANCE_NAME" | sha256sum --strict --check --status || exit -2
```

Comment:
@laurentsimon I copied this from the Go builder's code but I wonder, is this strictly necessary? The provenance is signed so if it's modified at any point it would be detected by the verifier. A malicious user could modify it at any point after the workflow completes as well so I don't think this provides any additional security.

Reply:
You're correct. It's not strictly needed.

Reply:
Ok, so just to be clear I'm going to leave it as is, but I'm not necessarily going to write an e2e test to test it.